huxley/api/serializers/user.py

```diff
@@ -7,21 +7,39 @@
 
 from datetime import datetime
 
+from django.core.mail import send_mail
+
 from huxley.accounts.models import User, School
 from huxley.api import validators
 from huxley.api.serializers.school import SchoolSerializer
 
 
 class UserSerializer(ModelSerializer):
-    school = SchoolSerializer(required=False, read_only=True)
+    school = SchoolSerializer(required=False)
 
     class Meta:
         model = User
         fields = ('id', 'username', 'first_name', 'last_name', 'user_type',
                   'school', 'committee',)
-        read_only_fields = ('id', 'username', 'first_name', 'last_name',
-                            'user_type', 'committee',)
+        read_only_fields = ('id', 'username', 'user_type', 'committee',)
 
+    def update(self, instance, validated_data):
+        if validated_data.get('school'):
+            school_data = validated_data.pop('school')
+            School.objects.filter(name=instance.school).update(**school_data)
+            User.objects.filter(username=instance).update(**validated_data)
```

Comment:
Same as above. Also, `instance` seems to suggest that it's an instance of a User... so why are we re-fetching the user? (Although that's a moot comment since I think we don't need to manually update).